js/form_validator.js

```diff
@@ -0,0 +1,53 @@
+// show a message with a type of the input
+function showMessage(input, message, type) {
+  // get the small element and set the message
+  const msg = input.parentNode.querySelector('small');
+  msg.innerText = message;
+  // update the class for the input
+  input.className = type ? 'email-success' : 'email-error';
+  return type;
+}
+
+function showError(input, message) {
+  return showMessage(input, message, false);
+}
+
+function showSuccess(input) {
+  return showMessage(input, '', true);
+}
+
+function hasValue(input, message) {
+  if (input.value.trim() === '') {
+    return showError(input, message);
+  }
+  return showSuccess(input);
+}
+
+function validateEmail(input, requiredMsg, invalidMsg) {
+  // check if the value is not empty
+  if (!hasValue(input, requiredMsg)) {
+    return false;
+  }
+  // eslint-disable-next-line
+  const emailRegex = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+  const email = input.value.trim();
+  if (!emailRegex.test(email)) {
+    return showError(input, invalidMsg);
+    // check if email is in small letter;
+  } if (email.toLowerCase() === email) {
+    return true;
+  }
+  return showError(input, invalidMsg);
+}
+
+const form = document.querySelector('#contact-form');
+const EMAIL_REQUIRED = 'Please enter your email';
+const EMAIL_INVALID = 'Please enter a correct email address format !';
+
+form.addEventListener('submit', (event) => {
+  // validate the form
+  const emailValid = validateEmail(form.elements['.email'], EMAIL_REQUIRED, EMAIL_INVALID);
+  if (!emailValid) {
+    event.preventDefault();
+  }
+});
```

Comment:
As requested by the project requirements:
![image](https://user-images.githubusercontent.com/42799579/202764397-c03524ef-8d03-4fb6-ba19-8a959163e87b.png)

Your form shouldn't allow upper-case letters 😉.
Currently, I was able to send the following message:
![image](https://user-images.githubusercontent.com/42799579/202764479-e0258242-c6bd-4b51-a6e8-014dbbfcd862.png)

Without receiving any custom error message. Please make sure to validate the user email.

Tip: This error is being displayed on the console:
![image](https://user-images.githubusercontent.com/42799579/202764573-c844f6bf-a39b-4c79-a012-25a6572d082e.png)

Make sure to check all the values being passed to your functions, looks like an `undefined` value is going through the `hasValue()` 😄.